src/Sentry/Protocol/ProfileInfo.cs

```diff
@@ -0,0 +1,90 @@
+using Sentry.Extensibility;
+using Sentry.Internal;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol;
+
+/// <summary>
+/// Profiling context information.
+/// </summary>
+internal sealed class ProfileInfo : IJsonSerializable
+{
+    /// <summary>
+    /// Profile's event ID.
+    /// </summary>
+    public SentryId EventId { get; private set; } = SentryId.Create();
+
+    public DebugMeta DebugMeta { get; set; } = new() { Images = new() };
+
+    private readonly Contexts _contexts = new();
+
+    /// <inheritdoc />
+    public Contexts Contexts
+    {
+        get => _contexts;
+        set => _contexts.ReplaceWith(value);
+    }
+
+    public SampleProfile Profile { get; set; } = new();
+
+    public DateTimeOffset StartTimestamp { get; set; } = DateTimeOffset.UtcNow;
+
+    public string? Environment { get; set; }
+
+    public string? Platform { get; set; } = Constants.Platform;
+
+    public string? Release { get; set; }
+
+    public Transaction? Transaction { get; set; }
+
+
+    public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
```

Comment:
So much duplication here with the serialization we already have on each context class.  There's also a lot more information available there, not being serialized here.  Is that intentional?

Reply:
Unfortunately yes, the deserialization on ingest is implemented manually and doesn't match these contexts. For example, OS name and version are required values while in our OS context, we don't set them at all (only raw description is set). Without these values, the profile will be dropped during ingest because it will raise an error when deserializing.